### ISSUE TITLE
Problem with roundToNearestMinutes function

### ISSUE THREAD

**Issue by benawhite** [date-fns/date-fns#3649]
There is an issue with the [roundToNearestMinutes function](https://date-fns.org/v2.30.0/docs/roundToNearestMinutes)

  ## Adds too many minutes to input date

  ### 💻 Code demonstrating the problem

  ```js
  import { roundToNearestMinutes } from 'date-fns'
  const dtm = new Date(2024, 1, 5, 10, 54);
  const defaultStartDate = roundToNearestMinutes(dtm, {
      nearestTo: 15,
      roundingMethod: 'ceil',
  });
  //=> 2024-01-05T11:15
  ```

  ### 🙁 Actual behavior

  input: 2024-01-05 10:54
  output: 2024-01-05 11:15

  ### 🙂 Expected behavior

  input: 2024-01-05 10:54
  output: 2024-01-05 11:00

  ## Debug information

  - date-fns version: 2.30.0
  - Browser/Node.js version: Chrome 120.0.6099.131 (Official Build) (64-bit)
  - Your timezone: America/New_York
  - Your current time: 11:36 AM

**Comment by jaymehta002** [comment]
hey i would like to suggest to use rounding function

instead of this
`const addedMinutes = Math.round(remainderMinutes / nearestTo) * nearestTo;`

use this
`const addedMinutes = roundingMethod(remainderMinutes / nearestTo) * nearestTo;`

i will also create a PR for correct code.

**Comment by fturmel** [comment]
This should be fixed in [3.2.0](https://github.com/date-fns/date-fns/releases/tag/v3.2.0)
